code/game/objects/structures/tables_racks.dm

```diff
@@ -224,24 +224,14 @@
 		deconstruct(TRUE)
 	return ITEM_INTERACT_SUCCESS
 
-/obj/structure/table/item_interaction_secondary(mob/living/user, obj/item/tool, list/modifiers)
-	if(istype(tool, /obj/item/construction/rcd))
-		return NONE
-
-	var/deck_act_value = NONE
-	if(istype(tool, /obj/item/toy/cards/deck))
-		deck_act_value = deck_act(user, tool, modifiers, TRUE)
-	// Continue to placing if we don't do anything else
-	if(deck_act_value != NONE)
-		return deck_act_value
-
-	if(!user.combat_mode)
-		return table_place_act(user, tool, modifiers)
-
-	return NONE
+// This extends base item interaction because tables default to blocking 99% of interactions
+/obj/structure/table/base_item_interaction(mob/living/user, obj/item/tool, list/modifiers)
+	. = ..()
+	if(.)
+		return .
 
-/obj/structure/table/item_interaction(mob/living/user, obj/item/tool, list/modifiers)
-	. = NONE
+	if(istype(tool, /obj/item/toy/cards/deck) && LAZYACCESS(modifiers, RIGHT_CLICK))
+		. = deck_act(user, tool, modifiers, TRUE)
```

Comment:
if we're moving this to the same item interaction proc as left clicking with a deck, we now have both
```dm
if(istype(tool, /obj/item/toy/cards/deck) && LAZYACCESS(modifiers, RIGHT_CLICK))
		. = deck_act(user, tool, modifiers, TRUE)
```
and
```dm
else if(istype(tool, /obj/item/toy/cards/deck))
		. = deck_act(user, tool, modifiers, FALSE)
```
in the same proc
where the only difference is that final boolean parameter, that's `TRUE` when `LAZYACCESS(modifiers, RIGHT_CLICK)` and `FALSE` when not

i feel it might be nicer to merge these in some way, rather than having the same typecheck but twice